### ISSUE TITLE
Delay in iOS confirm subscription pop-up screen

### ISSUE THREAD

**Issue by hariskh80** [chirag04/react-native-in-app-utils#160]
We are facing a strange issue. 

When InAppUtils.purchaseProduct is called, apple confirmation pop-up screen appears after 5-10 mins even if we leave the app. There is this long long delay in pop-up. We face this issue when we install our app via testflight to test it.

When we install the app from Xcode or run on simulator then there is no delay.

**Comment by chirag04** [comment]
interesting. I haven't seen anything like this before. All i can think of to debug is:

1) were you using the same itunes account when purchasing via testflight and on simulator?
2) Could it be app store issues? Can you test purchasing via simulator and testflight about the same time to make sure it's not app store.

**Comment by hariskh80** [comment]
Yes we have used similar and different itunes acocunt. Tried with accounts from different countries.

We have been facing this issues for days now so I doubt its to do with app store. 

We are baffled too :-)

**Comment by chirag04** [comment]
I don't have much to add here. I'm as surprised as you are. Try creating a new sandbox account maybe.

Unrelated but still: https://stackoverflow.com/questions/46257633/ios-11-is-in-app-purchase-testing-using-a-sandbox-user-keeps-asking-to-sign-in

**Comment by hugoh1995** [comment]
I have the same issue. I also starting to get this error message recently after 20 to 30 secondes passed after the buyProduct method is sent: 

```
code: "ENSURLERRORDOMAIN-1001"
domain: "NSURLErrorDomain"
message: "Connexion à l’iTunes Store impossible"
```

any idea ?

**Comment by chirag04** [comment]
refer this: https://stackoverflow.com/questions/6778167/undocumented-nsurlerrordomain-error-codes-1001-1003-and-1004-using-storeki

`1001` means request timedout. Possibly because the device was not connected to the internet or app store/sandbox was down.

**Comment by hariskh80** [comment]
To us it’s constantly happening and we did submit the app to test maybe it will be fixed on live app but still same issue

**Comment by Vilitsa** [comment]
Did you solve problem?

**Comment by hariskh80** [comment]
Nops but we submitted it anyway and on live app it was working fine.. strange

Regards,

Haris

**Comment by Vilitsa** [comment]
Thks

**Comment by superandrew213** [comment]
@hariskh80 try creating a new sandbox user and test how long it takes.

I have a feeling that the issue is linked to the number of transactions your testing account has made. The more transactions you have made the longer it will take to load them.

**Comment by hariskh80** [comment]
Hmm ok thanks

Regards,

Haris
